src/lib.rs

```diff
@@ -4,26 +4,329 @@
+    pub fn remove_recipient(id: RecipientId) {
+        <Recipients<T>>::remove(&id);
+    }
+
+    // Adds a new RewardRelationship to rewardRelationships, for a given source, recipient, account, etc., with identifier equal to current nextRewardRelationshipId. Also increments nextRewardRelationshipId.
+    pub fn add_reward_relationship(
+        mint_id: MintId,
+        recipient: RecipientId,
+        account: T::AccountId,
+        amount_per_payout: BalanceOf<T>,
+        next_payment_schedule: Option<NextPaymentSchedule<T::BlockNumber>>,
+        payout_interval: Option<T::BlockNumber>,
+    ) -> Result<RewardRelationshipId, RewardsError> {
+        ensure!(
+            <minting::Module<T>>::mint_exists(mint_id),
+            RewardsError::RewardSourceNotFound
+        );
+
+        let next_payment_at_block = match next_payment_schedule {
+            Some(schedule) => match schedule {
+                NextPaymentSchedule::Absolute(blocknumber) => {
+                    ensure!(
+                        blocknumber > <system::Module<T>>::block_number(),
+                        RewardsError::BlockNumberInPast
+                    );
+                    Some(blocknumber)
+                }
+                NextPaymentSchedule::Relative(blocknumber) => {
+                    Some(<system::Module<T>>::block_number() + blocknumber)
+                }
+            },
+            None => match payout_interval {
+                Some(interval) => Some(<system::Module<T>>::block_number() + interval),
+                None => {
+                    // No payouts will be made unless relationship is updated in future and next_payment_in_block
+                    // is set! should this be allowed?
+                    None
+                }
+            },
+        };
+
+        let relationship = RewardRelationship {
+            mint_id,
+            recipient,
+            account,
+            amount_per_payout,
+            next_payment_at_block,
+            payout_interval,
+            total_reward_received: Zero::zero(),
+            total_reward_missed: Zero::zero(),
+        };
+
+        let relationship_id = Self::next_reward_relationship_id();
+        NextRewardRelationshipId::mutate(|n| {
+            *n += 1;
+        });
+        <RewardRelationships<T>>::insert(relationship_id, relationship);
+        Ok(relationship_id)
+    }
+
+    // Removes a mapping from reward relashionships based on given identifier.
+    pub fn remove_reward_relationship(id: RewardRelationshipId) {
+        <RewardRelationships<T>>::remove(&id);
+    }
+
+    // For reward relationship found with given identifier, new values can be set for
+    // account, payout, block number when next payout will be made and the new interval after
+    // the next scheduled payout. All values are optional, but updating values are combined in this
+    // single method to ensure atomic updates.
+    pub fn set_reward_relationship(
+        id: RewardRelationshipId,
```

Comment:
Can we update the parameter names of the parameters that use the option to signal whether to update or not, e.g. `new_account` and `new_amount_per_payout`.

Reply:
done